src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -189,7 +189,7 @@ protected void Initialize(ISchema schema, IServiceProvider serviceProvider, IEnu
                 {
                     if (this[name] == null)
                     {
-                        AddType(graphType, context);
+                        AddTypeIfNotRegistered(graphType, context);
```

Comment:
Well, unfortunately this isn't helping.  I did a little investigation and found that `GetSchemaTypes` is going to create instances for each of the types listed in `AdditionalTypes`.  That could be solved easily enough with `.Distinct()` but it does not cover the situation where an indirect reference to a type previously added via `RegisterType` has a separate reference from the subsequent one added via `RegisterType`.  I don't have time to look at it further at the moment.

Reply:
So might as well revert this line at this time.